src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -252,3 +259,22 @@ def _init_seen_tables(self):
 
     def _table_already_upgraded(self, target) -> bool:
         return target in self._seen_tables
+
+    def revert_migrated_tables(self, *, schema: str | None = None, table: str | None = None):
+        def scope_filter(cur_table: Table):
+            return (not schema or cur_table.database == schema) and (not table or cur_table.name == table)
+
+        upgraded_tables = [
+            cur_table
+            for cur_table in list(self._tc.snapshot())
+            if scope_filter(cur_table) and cur_table.upgraded_to is not None
+        ]
+        for upgraded_table in upgraded_tables:
+            logger.info(
+                f"Reverting Table {upgraded_table.database}.{upgraded_table.name} "
+                f"upgraded_to {upgraded_table.upgraded_to}"
+            )
+            self._backend.execute(upgraded_table.sql_unset_to("hive_metastore"))
```

Comment:
and what happens to UC table/schema?

Reply:
I don't understand.

Reply:
we'd have a corrupt state - HMS thinks table is not migrated, but UC has it migrated already. Incorrect behavior.

Perhaps tune the logic the following way:

- If table is managed, prevent any reverts
- if table is external, prompt confirmation before we remove the table and update table properties

Reply:
I would have the following:
External tables - prompt and remove.
Managed tables - have the option to remove or not remove. Managed tables are disconnected from the original and potentially can be migrated multiple times. We can allow the customer to revert and re migrate without deleting it.